File: AGENDLY_APP/lib/features/clients/infrastructure/mappers/client_mapper.dart
import 'package:agendly_app/features/clients/domain/entities/client.dart';

class ClientMapper {
  static Client fromJson(Map<String, dynamic> json) {
    return Client(
      id: json['_id'] ?? json['id'],
      businessId: json['businessId'] ?? '',
      name: json['name'] ?? '',
      email: json['email'] ?? '',
      phone: json['phone'],
      address: json['address'],
      notes: json['notes'],
      source: json['source'] ?? 'manual',
      isActive: json['isActive'] ?? true,
      createdAt: DateTime.parse(json['createdAt'] ?? DateTime.now().toIso8601String()),
      updatedAt: DateTime.parse(json['updatedAt'] ?? DateTime.now().toIso8601String()),
    );
  }

  static Map<String, dynamic> toJson(Client client) {
    return {
      'businessId': client.businessId,
      'name': client.name,
      'email': client.email,
      'phone': client.phone,
      'address': client.address,
      'notes': client.notes,
      'source': client.source,
    };
  }
}

File: BACK/src/infrastructure/mappers/client.mapper.ts
import { ClientEntity } from '../../domain/entities/client.entity';
import { IClient } from '../datasources/mongo/models/client.model';

export class ClientMapper {
  static toEntity(model: IClient): ClientEntity {
    return {
      id: model._id.toString(),
      businessId: model.businessId,
      userId: model.userId,
      name: model.name,
      email: model.email,
      phone: model.phone,
      birthDate: model.birthDate,
      referredBy: model.referredBy,
      source: model.source as ClientEntity['source'],
      avatarUrl: model.avatarUrl,
      notes: model.notes,
      isActive: model.isActive,
      createdAt: model.createdAt,
      updatedAt: model.updatedAt,
    };
  }
}
